### PR TITLE
lower library verbosity

### PR DESCRIPTION
**Pull request by talg2324**
1. print(...) -> logging.log(logging.INFO, ...)
2. warnings.warn(...) -> logging.log(logging.WARN, ...)
3. removed linux=True
4. os.system -> subprocess.run
5. Option to disconnect subprocess pipe and hide executable console logging

### CODE REVIEW

File: kwave/options/simulation_execution_options.py [hunk 1]
```diff
@@ -34,6 +34,9 @@ class SimulationExecutionOptions:
     system_call: Optional[str] = None
     verbose_level: int = 0
 
+    # show simulation log
+    show_sim_log: bool = True
```
Comment (faridyagubbayli) [inline kwave/options/simulation_execution_options.py:38]: This variable is not used, what's the reason for introducing it?

File: tests/setup_test.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
+import logging
 import sys
 import os
 
 
 PACKAGE_PARENT = '..'
 SCRIPT_DIR = os.path.dirname(os.path.realpath(os.path.join(os.getcwd(), os.path.expanduser(__file__))))
-print(os.path.normpath(os.path.join(SCRIPT_DIR, PACKAGE_PARENT)))
+logging.log(logging.INFO,  os.path.normpath(os.path.join(SCRIPT_DIR, PACKAGE_PARENT)))
```
Comment (faridyagubbayli) [inline tests/setup_test.py:8]: There are two spaces after `logging.log(logging.INFO,` part in all changes in this PR. Maybe consider using one space instead?

**Review by faridyagubbayli — APPROVED** [review]
LGTM! Left two non-blocking comments. Thanks for working on this!

File: kwave/options/simulation_execution_options.py (hunk 1 above)
Comment (talg2324) [reply kwave/options/simulation_execution_options.py:38]: used in line 31 of executor.py to pipe the stdout to DEVNULL

File: tests/setup_test.py (hunk 2 above)
Comment (talg2324) [reply tests/setup_test.py:8]: good catch, I'll fix. Thanks

File: kwave/options/simulation_execution_options.py (hunk 1 above)
Comment (faridyagubbayli) [reply kwave/options/simulation_execution_options.py:38]: Ahh okay thanks!

**Review by waltsims — CHANGES_REQUESTED** [review]
This looks great. Thank you.

I have left a few non blocking comments. There one breaking changes in the CI, but other than that, this can be merged in.

**Comment by waltsims** [comment]
Also if you merge master into your PR, the CI will run for the PR as well.

File: kwave/utils/mapgen.py [hunk 3]
```diff
@@ -337,7 +337,7 @@ def power_law_kramers_kronig(w: np.ndarray, w0: float, c0: float, a0: float, y:
     """
 
     if 0 >= y or y >= 3:
-        warnings.warn("y must be within the interval (0,3)", UserWarning)
+        logging.log(logging.WARN, f"{UserWarning.__name__}: y must be within the interval (0,3)")
```
[the remark this answers is not in the document]
Comment (talg2324) [reply kwave/utils/mapgen.py:340]: Sure. It looks a bit awkward.. not sure that pytests can be adapted to work with logging module so maybe this is a long-term advantage to using warnings.warn. 
Should be possible to read the last logging message from the root logger (or module-specific logger when implemented) instead, but that requires sorting the module logger first

**Comment by talg2324** [comment]
Hi, I think I've addressed all your comments.
The only one I didn't implement was Walter's comment on adding the namespace to the logger.
I think it's a good idea, but I'll leave that one to you guys because there's several good ways to implement+configure it and I'm sure you have something in mind.

**Review by waltsims — APPROVED** [review]
Thanks for those changes. Can you please correct the RUFF formatting errors?

once complete we can merge this in:
```
Error: kwave/reconstruction/beamform.py:188:141: E501 Line too long (143 > 140 characters)
Error: tests/diff_utils.py:2:20: F401 `pprint.pprint` imported but unused
```

**Comment by talg2324** [comment]
> Thanks for those changes. Can you please correct the ruff formatting errors?
> 
> once complete we can merge this in:
> 
> ```
> Error: kwave/reconstruction/beamform.py:188:141: E501 Line too long (143 > 140 characters)
> Error: tests/diff_utils.py:2:20: F401 `pprint.pprint` imported but unused
> ```

Can you approve the test workflow?

**Comment by waltsims** [comment]
Thanks for the great PR. I've merged this in.

I hope you contribute again.

Best,
Walter

**Pull request merged**